### PR TITLE
add test framework and other minor changes

### PR DESCRIPTION
**Pull request by killpanda**
* rename logging -> update_annotations

### CODE REVIEW

File: flask_zipkin.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def _before_request(self):
             return
         headers = request.headers
         trace_id = headers.get('X-B3-TraceId') or self._gen_random_id()
-        parent_span_id = headers.get('X-B3-Parentspanid')
+        parent_span_id = headers.get('X-B3-ParentSpanId')
```
Comment (GuoJing) [inline flask_zipkin.py]: Yes! This is what wangbing said "we are not pro, blablabla"

File: flask_zipkin.py [hunk 2]
```diff
@@ -123,6 +123,10 @@ def create_http_headers_for_new_span(self):
         return zipkin.create_http_headers_for_new_span()
 
     def logging(self, **kwargs):
-        if hasattr(g, '_zipkin_span') and g._zipkin_span and g._zipkin_span.logging_context:
```
Comment (GuoJing) [inline flask_zipkin.py]: warn('this method is deprecated, try to use xxx method')

Comment (killpanda) [inline flask_zipkin.py]: 加上了

**Comment by killpanda** [comment]
shall we merge this?

File: flask_zipkin.py [hunk 3]
```diff
@@ -123,6 +124,13 @@ def create_http_headers_for_new_span(self):
...
+        logging.warning('This method has been depreated, '
+                        'please call `update_annotations` instead.')
+        self.update_annotations(**kwargs)
+
+    def update_annotations(self, **kwargs):
```
Comment (adriancole) [inline flask_zipkin.py]: The term annotation is most often used for annotations in the zipkin model (timestamp -> string)
if you want a shorter word for binary_annotations, I'd use "update_tags"

for example, we call kv tag in brave.

https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/Span.java#L88

**Comment by GuoJing** [comment]
update_tags is better I think.

Thanks!
------
http://www.guojing.me

2017年1月15日 +0800 AM10:16 Adrian Cole <notifications@github.com>，写道：

File: flask_zipkin.py (hunk 1 above)
Comment (natlownes) [inline flask_zipkin.py]: oh, cool.  didn't notice this when I did my pull request.  I'll close mine and wait for this to be merged

**Comment by GuoJing** [comment]
Shall we merge this? @killpanda

**Comment by GuoJing** [comment]
@killpanda shall we merge this?

**Comment by killpanda** [comment]
sorry for being late. plz review.

**Comment by GuoJing** [comment]
LGTM

**Pull request merged**